internal/pipe/ko/ko.go

```diff
@@ -0,0 +1,301 @@
+// Package ko implements the pipe interface with the intent of
+// building OCI compliant images with ko.
+package ko
```

Comment:
Sort of a high level comment: a lot of this code seems to be copied from `ko`, which makes me wonder if we might want to refactor `ko` to make it easier to reuse, instead of copying. I don't have any more specific comments, just something to consider. As `ko` changes its version of this code, it might get out of sync with goreleaser's and cause confusion or maintenance burden to keep features in sync.

Reply:
I've actually copied these codes from the [terraform-provider-ko](https://github.com/chainguard-dev/terraform-provider-ko/blob/main/internal/provider/resource_ko_image.go#L180) project.